src/System.IO.Compression/src/System/IO/Compression/ZipArchiveEntry.Unix.cs

```diff
@@ -31,8 +32,31 @@ internal static string ParseFileName(string path, ZipVersionMadeByPlatform madeB
             }
             else
             {
+                PathInternal.CheckInvalidPathChars(path);
                 return Path.GetFileName(path);
             }
         }
+
+        internal static readonly char[] InvalidPathChars =
```

Comment:
I'd like to avoid making existing projects add any new Common files so that restricts me to adding the CheckInvalidChars_Windows/Unix functions to PathInternal.cs since it's already included and isn't platform-specific. 

So the design is going to look like this:

PathInternal.cs:
- Contains public CheckInvalidChars_Windows, CheckInvalidChars_Unix

PathInternal.Windows.cs and PathInternal.Unix.cs
- Both are part of partial class PathInternal
- Contain public function CheckInvalidChars that consists entirely of a call to CheckInvalidChars_Windows or CheckInvalidChars_Unix

This way no existing code needs to be changed. CheckInvalidChars functions as it currently does. The CheckInvalidChars_OSGROUP functions can be accessed by PathInternal.CheckInvalidChars_Windows etc.

I could instead add new classes PathInternal_Windows and PathInternal_Unix to PathInternal.cs but I it's more useful imo to put them into the PathInternal class.